src/type_icons.c

```diff
@@ -0,0 +1,508 @@
+{
+    .data = gBattleIcons_Pal1,
+    .tag = TYPE_ICON_TAG
+};
+
+const struct CompressedSpritePalette sTypeIconPal2 =
+{
+    .data = gBattleIcons_Pal2,
+    .tag = TYPE_ICON_TAG_2
+};
+
+const struct OamData sOamData_TypeIcons =
+{
+    .affineMode = ST_OAM_AFFINE_OFF,
+    .objMode = ST_OAM_OBJ_NORMAL,
+    .shape = SPRITE_SHAPE(8x16),
+    .size = SPRITE_SIZE(8x16),
+    .priority = 1,
+};
+
+const struct CompressedSpriteSheet sSpriteSheet_TypeIcons2 =
+{
+    .data = gBattleIcons_Gfx2,
+    .size = (8*16) * 9,
+    .tag = TYPE_ICON_TAG_2,
+};
+
+const struct CompressedSpriteSheet sSpriteSheet_TypeIcons1 =
+{
+    .data = gBattleIcons_Gfx1,
+    .size = (8*16) * 10,
+    .tag = TYPE_ICON_TAG,
+};
+
+const struct SpriteTemplate sSpriteTemplate_TypeIcons1 =
+{
+    .tileTag = TYPE_ICON_TAG,
+    .paletteTag = TYPE_ICON_TAG,
+    .oam = &sOamData_TypeIcons,
+    .anims = sSpriteAnimTable_TypeIcons,
+    .images = NULL,
+    .affineAnims = gDummySpriteAffineAnimTable,
+    .callback = SpriteCB_TypeIcon
+};
+
+const struct SpriteTemplate sSpriteTemplate_TypeIcons2 =
+{
+    .tileTag = TYPE_ICON_TAG_2,
+    .paletteTag = TYPE_ICON_TAG_2,
+    .oam = &sOamData_TypeIcons,
+    .anims = sSpriteAnimTable_TypeIcons,
+    .images = NULL,
+    .affineAnims = gDummySpriteAffineAnimTable,
+    .callback = SpriteCB_TypeIcon
+};
+
+void LoadTypeIcons(u32 battler)
+{
+    u32 position;
+
+    if (B_SHOW_TYPES == SHOW_TYPES_NEVER)
+        return;
+
+    LoadTypeSpritesAndPalettes();
+
+    for (position = 0; position < gBattlersCount; ++position)
+        LoadTypeIconsPerBattler(battler, position);
+}
+
+static void LoadTypeSpritesAndPalettes(void)
+{
+    if (IndexOfSpritePaletteTag(TYPE_ICON_TAG) != UCHAR_MAX)
+        return;
+
+    LoadCompressedSpriteSheet(&sSpriteSheet_TypeIcons1);
+    LoadCompressedSpriteSheet(&sSpriteSheet_TypeIcons2);
+    LoadCompressedSpritePalette(&sTypeIconPal1);
+    LoadCompressedSpritePalette(&sTypeIconPal2);
+}
+
+static void LoadTypeIconsPerBattler(u32 battler, u32 position)
+{
+    u32 typeNum, types[2];
+    u32 battlerId = GetBattlerAtPosition(position);
+    bool32 useDoubleBattleCoords = UseDoubleBattleCoords(battlerId);
+
+    if (!IsBattlerAlive(battlerId))
+        return;
+
+    for (typeNum = 0; typeNum < 2; ++typeNum)
+        types[typeNum] = GetMonPublicType(battlerId, typeNum);
+
+    for (typeNum = 0; typeNum < 2; ++typeNum)
+        CreateSpriteFromType(position, useDoubleBattleCoords, types, typeNum, battler);
+}
+
+static bool32 UseDoubleBattleCoords(u32 position)
+{
+    if (!IsDoubleBattle())
+        return FALSE;
+
+    if ((position == B_POSITION_PLAYER_LEFT) && (gBattleMons[B_POSITION_PLAYER_RIGHT].species == SPECIES_NONE))
+        return FALSE;
+
+    if ((position == B_POSITION_OPPONENT_LEFT) && (gBattleMons[B_POSITION_OPPONENT_RIGHT].species == SPECIES_NONE))
+        return FALSE;
+
+    return TRUE;
+}
+
+static u32 GetMonPublicType(u32 battlerId, u32 typeNum)
```

Comment:
the content of this function has spacing issues

Reply:
https://github.com/rh-hideout/pokeemerald-expansion/pull/5131/commits/654844b9321a12eb1ebd1c918c3d5e86eca028fd